### PR TITLE
[13.0][ADD] stock_scrap_cancel: New addon

### PR DESCRIPTION
**Pull request by victoralmau**
This module adds the cancel status to the scraps and allows the canceled ones to be converted into a draft.

Please @chienandalu and @pedrobaeza can you review it?

@Tecnativa TT34293

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]

**Review by chienandalu — COMMENTED** [review]
I detected an issue on stock_lot_scrap: quick fix here: https://github.com/OCA/stock-logistics-workflow/pull/963 Anyway, nothing to do with this.

I've been testing and I find this behavior:

- I go to a reception picking and validate it.
- I get the pickings quantities in my stock.
- Then I go back to the picking and I scrap some units.
- My stock diminishes in the same amount of units as they're moved to the scrap location.
- Then I cancel the scrap and that move is erased.
- But my stock doesn't recover those units!

Shouldn't the stock be reverted from scrap to my inventory?

**Comment by victoralmau** [comment]
> I detected an issue on stock_lot_scrap: quick fix here: #963 Anyway, nothing to do with this.
> 
> I've been testing and I find this behavior:
> 
>     * I go to a reception picking and validate it.
> 
>     * I get the pickings quantities in my stock.
> 
>     * Then I go back to the picking and I scrap some units.
> 
>     * My stock diminishes in the same amount of units as they're moved to the scrap location.
> 
>     * Then I cancel the scrap and that move is erased.
> 
>     * But my stock doesn't recover those units!
> 
> 
> Shouldn't the stock be reverted from scrap to my inventory?

Thanks for the review.
You are right, the stock should be increased again when canceling. Can you check again that this last change is correct?

**Review by chienandalu — APPROVED** [review]
Great works like charm now :slightly_smiling_face: :+1:

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
We have to check why a manual move is needed instead of only amending quantity. cc @carlosdauden

**Comment by victoralmau** [comment]
> 

> We have to check why a manual move is needed instead of only amending quantity. cc @carlosdauden

I've applied a change to simplify everything which was probably what you had in mind.

File: stock_scrap_cancel/models/stock_scrap.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    _inherit = "stock.scrap"
+
+    state = fields.Selection(selection_add=([("cancel", "Cancelled")]))
+
+    def action_draft(self):
+        return self.write({"state": "draft"})
+
+    def action_cancel(self):
+        old_scrap_qty = self.scrap_qty
+        self.move_id.move_line_ids.write({"qty_done": 0.0})
```
Comment (pedrobaeza) [inline stock_scrap_cancel/models/stock_scrap.py:19]: You should unlink the move after doing this.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**